chore(tutorials): content review (#2980)
scaleway/docs-content: tutorials/install-apache-ansible/index.mdx

categories:
  - compute
dates:
  validation: 2024-03-26
  posted: 2018-08-02
---

> Apache is one of the most of open-source web servers globally. This tutorial explains how to deploy and configure Apache on Scaleway Instances with the power of Ansible automation. Before diving in, ensure you meet the prerequisites:

Reviewer: ```suggestion
Apache is one of the most used open-source web servers globally. This tutorial explains how to deploy and configure Apache on Scaleway Instances with the power of Ansible automation. Before diving in, ensure you meet the prerequisites:
```

Reviewer: Not sure which word was missing, so I added `used.`

<Macro id="requirements" />

> - Owner status or appropriate IAM permissions within your Scaleway Organization.

Reviewer: ```suggestion
- [Owner](/identity-and-access-management/iam/concepts/#owner) status or [IAM permissions](/identity-and-access-management/iam/concepts/#permission) allowing you to perform actions in the intended Organization
```

- Familiarity with [Ansible](/tutorials/ansible-bionic-beaver) for seamless execution.

## Configuring Ansible for Apache

Begin by setting up Ansible to communicate with your Scaleway Instances effectively:

1. Create a dedicated directory:
    ```
    mkdir ansible-apache
    ```
2. Navigate to the directory:
    ```
    cd ~/ansible-apache/
    ```
3. Create an `ansible.cfg` file:
    ```
    nano ansible.cfg
    ```

> 4. Populate it with hostfile configuration:

Reviewer: ```suggestion
4. Populate it with the host file configuration:
```

    ```
    [defaults]
    hostfile = hosts
    ```
5. Create and edit a `hosts` file:
    ```
    nano hosts
    ```
6. Define your host information:
    ```
    [apache]
    secondary_server_ip ansible_ssh_user=username
    ```

## Creating a Playbook

Utilize Ansible playbooks for configuration and deployment:

1. Craft a playbook named `apache.yml`:
    ```
    nano apache.yml
    ```
2. Insert the following content:
    ```
    - hosts: apache
      tasks:
        - name: run echo command
          command: /bin/echo hello world
    ```
3. Execute the playbook:
    ```
    ansible-playbook apache.yml
    ```

## Installing Apache

> Install Apache on your Scaleway Instance:

Reviewer: Should this first line be a step?

1. Update `apache.yml` with the apt module:
    ```
    nano apache.yml
    ```

> 2. Replace existing content with:

Reviewer: ```suggestion
2. Replace the existing content with:
```
